algorithms/src/snark/varuna/varuna.rs

```diff
@@ -310,11 +310,14 @@ where
         let evaluations_at_point = AHPForR1CS::<E::Fr, SM>::evaluate_index_polynomials(state, circuit_id, point)?;
         let mut lc = crate::polycommit::sonic_pc::LinearCombination::empty("circuit_check");
         let mut evaluation = E::Fr::zero();
+        ensure!(poly_info.len() == challenges.len() + 1);
```

Comment:
Directly check against `linear_combination_challenges`

Reply:
Refactored so we are assured the lengths are equal and we don't panic: fdfd68a9c